src/Compilers/CSharp/Test/Semantic/Semantics/InlineArrayTests.cs

```diff
@@ -768,12 +770,180 @@ void verify(ModuleSymbol m)
 
                 Assert.True(buffer.HasInlineArrayAttribute(out int length));
                 Assert.Equal(10, length);
-                Assert.Equal(SpecialType.System_Int32, buffer.TryGetInlineArrayElementType().SpecialType);
+                Assert.Equal(SpecialType.System_Int32, buffer.TryGetInlineArrayElementField().Type.SpecialType);
 
                 Assert.Equal("Field", m.GlobalNamespace.GetTypeMember("C1").GetAppliedConditionalSymbols().Single());
             }
         }
 
+        [Fact]
+        public void InlineArrayType_26()
+        {
+            var src = @"
+[System.Runtime.CompilerServices.InlineArray(10)]
+struct Buffer
+{
+    private int _element0;
+    public event System.Action E;
+}
+";
+            var comp = CreateCompilation(src + InlineArrayAttributeDefinition);
+            CompileAndVerify(comp, symbolValidator: verify, sourceSymbolValidator: verify).VerifyDiagnostics();
+
+            void verify(ModuleSymbol m)
+            {
+                var buffer = m.GlobalNamespace.GetTypeMember("Buffer");
+
+                Assert.True(buffer.HasInlineArrayAttribute(out int length));
```

Comment:
>Assert.True(buffer.HasInlineArrayAttribute(out int length)); [](http://example.com/codeflow?start=16&length=60)

Why is the type with two fields recognized as a valid inline array type?

Reply:
> Why is the type with two fields recognized as a valid inline array type?

It is not. This API is not determining whether the type is a valid inline array type. It only indicates whether it has a valid attribute. The final check is performed by ```TryGetInlineArrayElementField```, which returns the only field